test/gtest/ucm/malloc_hook.cc

```diff
@@ -406,6 +406,11 @@ void test_thread::test() {
     EXPECT_GT(total_ptrs_in_range, 0u);
 
     /* Allocate large chunk */
+#ifdef __GNUC__ // Check if the compiler is GNUC
```

Comment:
really dangerous hack which may hide real errors...
may be update function `is_ptr_in_range` like this:
```
bool is_ptr_in_range(void *ptr, size_t size, const std::vector<range> &ranges) {
    uintptr_t p = (uintptr_t)ptr;
    uintptr_t begin, end;

    for (std::vector<range>::const_iterator iter = ranges.begin(); iter != ranges.end(); ++iter) {
        begin = (uintptr_t)iter->first;
        end   = (uintptr_t)iter->second;
        if ((p >= begin) && (p < end)) {                                                                                                                              34    311                 return true;
        }
    }
    return false;
}
```

Reply:
@hoopoepg this is not as dangerous because it disables the warning only for a small subsection of code but it is not pretty for sure. I will try your suggestion.

Reply:
@hoopoepg your solution works as well. I will update the patch. thanks !